src/js/common/components/Pagination.jsx

```diff
@@ -24,43 +26,50 @@ class Pagination extends React.Component {
     let prevPage;
     if (this.props.page > 1) {
       prevPage = (
-        <a className="va-pagination-prev" onClick={() => {this.props.onPageSelect(this.props.page - 1);}}>
-          Previous
+        <a onClick={() => {this.props.onPageSelect(this.props.page - 1);}}>
+          <abbr title="Previous">Prev</abbr>
         </a>
       );
     }
     return prevPage;
   }
 
+  pageNumbers(limit) {
+    const totalPages = this.props.pages;
+    // If there are more pages returned than the limit to show
+    // cap the upper range at limit + the page number.
+    const end = (totalPages > limit) ? (limit + this.props.page) : totalPages;
+
+    return range(this.props.page, end);
+  }
 
   render() {
     if (this.props.pages === 1) {
       return <div/>;
     }
 
-    const pageList = Array(this.props.pages).fill().map((e, i) => {
-      const pageNumber = i + 1;
+    const pageList = this.pageNumbers(10).map((e) => {
```

Comment:
`e` is a pretty non-descriptive name. Can this be `pageNumber` or something now that it actually is the number?